layout/theme.liquid

```diff
@@ -47,11 +47,16 @@
             window.location.pathname !== '/throttle/queue' &&
             window.location.pathname !== '/challenge'
         ) {
-          var currentHostname = window.location.hostname;
-          var storefrontHostname = "{{ settings.storefront_hostname }}";
+          var currentUrl = new URL(window.location.href);
+          var storefrontUrl = new URL("{{ settings.storefront_hostname }}");
```

Comment:
I think this will throw in almost all cases because the hostname will have `https://` prefix stripped, is that right?